core/trino-main/src/main/java/io/trino/server/security/PasswordAuthenticator.java

```diff
@@ -65,6 +74,25 @@ public Identity authenticate(ContainerRequestContext request)
         }
     }
 
+    /**
+     * When the user in the basic authentication header matches the x-trino-user header, we assume that the client does
+     * not want to force the runtime user name, and only wanted to communicate the authentication user.
+     */
+    private void clearRequestUserHack(BasicAuthCredentials basicAuthCredentials, MultivaluedMap<String, String> headers)
```

Comment:
What hack is this referring to?

Reply:
I use the word `hack` here because explaining what we are doing takes an entire paragraph of text, so I couldn't come up with a reasonable name for the method.  I actually like the method name because it stands out in our code base because it is unusual, so someone will notice (like you did).

Reply:
I understand that you consider your implementation a hack, still giving it a proper name is useful.

Reply:
Actually, I changed this to rewrite the header, so protocol detection will continue to work unchanged.